### ISSUE TITLE
Replace fan motor

### ISSUE THREAD

**Issue by harrisjm3** [device-teardowns/coway-ap1512hh#3]
Thanks for the good info. Have you managed to find a replacement part for the fan motor? I have an older model, I think circa 2016 that has a Panasonic motor with similar specs. So I would assume I could replace it with the Nidec model. 

Would love some links to where others have purchased.

**Comment by larryqiann** [comment]
Hi, thank you for your comment. I too am having challenges sourcing these motors. Since this teardown was created, I procured another used unit locally, to use the motor for another application, which is older and contains a Panasonic motor. There are no other mechanical or electrical changes to the unit other than the brand of the motor. 

Both motors perform very similarly in noise characteristics. The unit will do closed loop speed control from 25-75% PWM duty cycle, so slight differences should not cause any issue

**Comment by harrisjm3** [comment]
![PXL_20220205_235034182 MP](https://user-images.githubusercontent.com/32304676/152712646-04df01b8-6b3e-4f0c-b116-a5eb73576822.jpg)

**Comment by larryqiann** [comment]
Yes, the picture you have posted is the same motor in both my units. 

I think the rubber grommets may or may not be compatible across motors, but the motor and grommet combination should be.

Are you experiencing issues with the Panasonic motor?

**Comment by harrisjm3** [comment]
Yes, when I run it on high the motor starts to squeal very loudly and the fan moves at different speeds, but overall slower than it should.

**Comment by larryqiann** [comment]
Does the motor squeal when turned by hand?

The bearings in these motors are actually changeable, with some effort, and are fairly standard in size. Normally, you would replace the motor, but i've only seen one on eBay in the recent past and you'd need to match the E clip distance and shaft length. The metal plate on top of the motor needs to be uniformly pried off with a wide flathead carefully, and the bearings replaced using a bearing puller.

**Comment by harrisjm3** [comment]
No, and it doesn't squeal at low and medium speed. Just on high.

Do you have a write up on replacing ball bearings? I've not done anything like that before.

**Comment by harrisjm3** [comment]
It also does not squeal on high if I take the fan off of the motor.

**Comment by larryqiann** [comment]
I do not currently have a writeup, though you might try looking up the replacement of bearings in mini split motors, which are the other application of similar motors.

Would you be able to post a video or audio clip of the sound? Thanks

**Comment by harrisjm3** [comment]
https://photos.app.goo.gl/BTHgPJ8LmryrHnyz8

**Comment by harrisjm3** [comment]
https://photos.app.goo.gl/SDtAZJUZg7jWRuit8

**Comment by harrisjm3** [comment]
I emailed Coway support and of course they reply stating that they do not sell internal replacement parts. Lame.

**Comment by larryqiann** [comment]
Hi,

Sorry about the late reply. I was outside and was not able to view the videos until now.

The sound that the device makes when the fan is attached is interesting - it does not actually sound to me like the bearings in the motor themselves are the problem, but rather that the bearing is either spinning in the case, or there's something, like a washer or something, that's loose on the fan.  usually when ball bearings go bad, it sounds like there are marbles rolling in the unit and the sound doesn't go away.

Does the shaft have significant play, either side to side, or forward and backwards?

**Comment by ohmantics** [comment]
As one of the people maintaining connectivity tools with the WiFi version (AP-1512HHS), I'm wondering what that version's control board looks like and ideally would love to see what the firmware in that version looks like to see if we can find a way to control it locally without using the Coway cloud servers.

**Comment by larryqiann** [comment]
> As one of the people maintaining connectivity tools with the WiFi version (AP-1512HHS), I'm wondering what that version's control board looks like and ideally would love to see what the firmware in that version looks like to see if we can find a way to control it locally without using the Coway cloud servers.

Hi there, 

Thank you for your comments! Unfortunately, I do not have this version of the device. I have seen it around online before, but currently I don't have plans to procure it. 

However, something that may be of interest to you is that I have designed and built 5 replacement control boards for these non-smart units. They have an ESP32-WROOM module on board and have the exact same connectors, mounting holes, buttons, and LEDs as the original. I have not had time to write any firmware for them (or test them) as I have been occupied since building the boards, but if you are interested, I can release/open source the hardware part if this will be of interest to you.

My best guess is that their design looks very similar to what I came up with - I would assume the power board section of the device remains the same, but a different control board is fitted. 

![CleanShot 2023-01-14 at 17 47 44](https://user-images.githubusercontent.com/25872705/212509388-71f680a3-dc60-40d1-9b67-d545d5946d08.png)

**Comment by ohmantics** [comment]
Very cool. Coway's model uses a Wiznet module for the wifi. https://www.wiznet.io/product-item/wizfi250/ https://fccid.io/XR2WIZFI250

**Comment by sg1888** [comment]
> > As one of the people maintaining connectivity tools with the WiFi version (AP-1512HHS), I'm wondering what that version's control board looks like and ideally would love to see what the firmware in that version looks like to see if we can find a way to control it locally without using the Coway cloud servers.
> 
> Hi there,
> 
> Thank you for your comments! Unfortunately, I do not have this version of the device. I have seen it around online before, but currently I don't have plans to procure it.
> 
> However, something that may be of interest to you is that I have designed and built 5 replacement control boards for these non-smart units. They have an ESP32-WROOM module on board and have the exact same connectors, mounting holes, buttons, and LEDs as the original. I have not had time to write any firmware for them (or test them) as I have been occupied since building the boards, but if you are interested, I can release/open source the hardware part if this will be of interest to you.
> 
> My best guess is that their design looks very similar to what I came up with - I would assume the power board section of the device remains the same, but a different control board is fitted.
> 
> ![CleanShot 2023-01-14 at 17 47 44](https://user-images.githubusercontent.com/25872705/212509388-71f680a3-dc60-40d1-9b67-d545d5946d08.png)

Very cool!  Did you ever get around to writing the firmware for the boards?  I'm curious how we could tap into the existing board to send commands to control the purifier.  I have a few of these Coways, and would love to make them "smart" without having to go out and buy new purifiers.
